specs/deneb/polynomial-commitments.md

```diff
@@ -338,24 +341,30 @@ def blob_to_kzg_commitment(blob: Blob) -> KZGCommitment:
     """
     Public method.
     """
+    assert len(blob) == BYTES_PER_BLOB
     return g1_lincomb(bit_reversal_permutation(KZG_SETUP_LAGRANGE), blob_to_polynomial(blob))
 ```
 
 #### `verify_kzg_proof`
 
 ```python
 def verify_kzg_proof(commitment_bytes: Bytes48,
-                     z: Bytes32,
-                     y: Bytes32,
+                     z_bytes: Bytes32,
+                     y_bytes: Bytes32,
                      proof_bytes: Bytes48) -> bool:
     """
     Verify KZG proof that ``p(z) == y`` where ``p(z)`` is the polynomial represented by ``polynomial_kzg``.
     Receives inputs as bytes.
     Public method.
     """
+    assert len(commitment_bytes) == BYTES_PER_COMMITMENT
+    assert len(z_bytes) == BYTES_PER_FIELD_ELEMENT
+    assert len(y_bytes) == BYTES_PER_FIELD_ELEMENT
+    assert len(proof_bytes) == BYTES_PER_PROOF
```

Comment:
You're right, typing *hint* doesn't mandatorily verify the length. You can also assert it is checking with SSZ `remerkleable`:

- You can cast: `Bytes48(commitment_bytes)`, it will raise an error if the length is wrong.
- You can use `assert isinstance(commitment_bytes, Bytes48)` to force the input to be `remerkleable`-defined Bytes48. That said, Python's built-in 48-byte object will be an invalid input.

However, I think the assertions you added are more explicit and clear for client devs. 👍

Reply:
Interesting, thanks! Sounds good 👍